builtin/logical/database/path_rotate_role_credentials.go

```diff
@@ -69,6 +60,37 @@ func (b *databaseBackend) pathRotateRoleCredentialsUpdate() framework.OperationF
 	}
 }
 
+func (b *databaseBackend) populateRotationQueue(ctx context.Context, s logical.Storage) {
+	log := b.Logger()
+
+	b.Lock()
+	defer b.Unlock()
+
+	roles, err := s.List(ctx, "role/")
```

Comment:
Processing the list of roles needs to happen in a goroutine, to avoid blocking the remaining initialization of the Backend.